be/src/pipeline/exec/join/process_hash_table_probe_impl.h

```diff
@@ -70,7 +70,9 @@ ProcessHashTableProbe<JoinOpType>::ProcessHashTableProbe(HashJoinProbeLocalState
           _build_side_output_timer(parent->_build_side_output_timer),
           _probe_side_output_timer(parent->_probe_side_output_timer),
           _finish_probe_phase_timer(parent->_finish_probe_phase_timer),
-          _right_col_idx((_parent_operator->_is_right_semi_anti && !_have_other_join_conjunct)
+          _right_col_idx((_parent_operator->_is_right_semi_anti && !_have_other_join_conjunct &&
```

Comment:
better directly get right_col_idx from parent operator, and use `left+right=intermediate` to do double check